openfisca_core/tools/__init__.py

```diff
@@ -15,14 +16,21 @@ def assert_near(value, target_value, absolute_error_margin = None, message = '',
     if isinstance(message, unicode):
         message = message.encode('utf-8')
     if isinstance(value, np.ndarray):
-        if absolute_error_margin is not None:
-            assert (abs(target_value - value) <= absolute_error_margin).all(), \
-                '{}{} differs from {} with an absolute margin {} > {}'.format(message, value, target_value,
-                    abs(target_value - value), absolute_error_margin)
-        if relative_error_margin is not None:
-            assert (abs(target_value - value) <= abs(relative_error_margin * target_value)).all(), \
-                '{}{} differs from {} with a relative margin {} > {}'.format(message, value, target_value,
-                    abs(target_value - value), abs(relative_error_margin * target_value))
+        if isinstance(target_value, Enum) or (isinstance(target_value, np.ndarray) and target_value.dtype == object):
+            if not value.dtype == object:
+                assert False("Expected a Enum, got {} of dtype {}".format(value, value.dtype))
+            else:
+                assert (target_value == value).all(), "Expected {}, got {}".format(target_value, value)
+        else:
+
+            if absolute_error_margin is not None:
+                assert (abs(target_value - value) <= absolute_error_margin).all(), \
```

Comment:
Seems to raise a `TypeError` for dates (as example of object).
My test is a bit simple so it might come from that: 
```
simulation = new_simulation(TEST_CASE)  # from test_entities.py    
test_date = simulation.household.members('variable_date', simulation.period)
assert_near(test_date, test_date)
```
where `test_date` is similar to `date_naissance`.